CMakeLists.txt

```diff
@@ -21,6 +24,15 @@ if(MSVC)
   endif()
 else()
   message("Setting warning flags")
+
+  check_cxx_compiler_flag(-Wold-style-cast WOLD_STYLE_CAST)
```

Comment:
I am not really a CMake expert :), could you shortly describe me how this works?
So if -Wold-style-cast is set, you set it? I am slightly confused why is first repeated as symbol and then as string, and what is WOLD_STYLE_CAST important for, and how this all works actually hm.

Maybe you check if that flag is supported by compiled, then result of this check gets set into WOLD_STYLE_CAST, and then you set it as actual compiler flag if that is so?
If so, why do you think we need to be so thorough, how likely is it that compiler will not support it?

Reply:
In general, you should test whether the flag works, and if it works, append it. In practice, only Clang and GCC support this flag. For instance, `icpc` (Intel's C++ compiler) does not and will bomb out on it. By testing flags, you maximize compiler compatibility across Unix compilers.

Reply:
Makes sense!